src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -227,6 +316,8 @@ class CoreWorkerDirectActorTaskReceiver : public rpc::DirectActorHandler {
   TaskHandler task_handler_;
   /// The IO event loop for running tasks on.
   boost::asio::io_service &task_main_io_service_;
+  /// Shared waiter for dependencies required by incoming tasks.
+  std::unique_ptr<DependencyWaiterImpl> waiter_;
```

Comment:
Does this need to be a unique_ptr?

Reply:
Yeah, you can't declare a polymorphic class inline.